packages/lists/src/components/FixedList/FixedList.tsx

```diff
@@ -0,0 +1,164 @@
+export interface IFixedListProps {
+  /**
+   * The total number of items contained in the list.
+   */
+  itemCount: number;
+
+  /**
+   * The fixed height of each item in the list.
+   */
+  itemHeight: number;
+
+  /**
+   * The current viewport state.
+   */
+  viewportState: IViewportState;
+
+  /**
+   * The height of the viewport this list is mounted in.
+   */
+  viewportHeight: number;
+
+  /**
+   * The width of the viewport this list is mounted in.
+   */
+  viewportWidth: number;
+
+  /**
+   * The distance of the top of the viewport surface to the top of the list surface.
+   */
+  surfaceTop: number;
+
+  /**
+   * The height of item overscan before and after the visible area of the viewport.
+   */
+  overscanHeight: number;
+
+  /**
+   * Callback used to render an item with the given index.
+   */
+  onRenderItem: (itemIndex: number, style: React.CSSProperties) => JSX.Element | null;
+
+  /**
+   * Callback used to add and modify the list's calculated materialized range, for example in order to always render
+   * a focused item, no matter whether it is currently in view or not.
+   */
+  onGetMaterializedRanges?: GetMaterializedRangesCallback;
+}
+
+export interface IItemRange {
+  startIndex: number;
+  endIndex: number;
+}
+
+export type GetMaterializedRangesCallback = (defaultMaterializedRanges: IItemRange[]) => IItemRange[];
+
+const TRAILING_OVERSCAN_COUNT_WHILE_SCROLLING = 1;
+
+/**
+ * Calculates the currently visible range of items based on the viewport state.
+ * @param props The FixedList props
+ * @return
+ */
+function getVisibleItemRange(props: IFixedListProps): IItemRange {
+  const { surfaceTop, itemHeight, viewportState, viewportHeight, itemCount } = props;
+
+  const scrollTop = viewportState.scrollDistance[Axis.Y] - surfaceTop;
+
+  const startIndex = Math.max(0, Math.floor(scrollTop / itemHeight));
+  const endIndex = Math.min(itemCount - 1, Math.ceil((scrollTop + viewportHeight) / itemHeight));
+
+  return { startIndex, endIndex };
+}
+
+/**
+ * Calculates the materialized range of items based on the visible range. Consumers of FixedList can provide
+ * a callback to add and modify the calculated materialized range, for example in order to always render
+ * a focused item, no matter whether it is currently in view or not.
+ * @param props The FixedList props
+ */
+function getMaterializedItemRanges(props: IFixedListProps): IItemRange[] {
+  const { viewportState, overscanHeight, onGetMaterializedRanges, itemHeight, itemCount } = props;
+
+  const { isScrolling, scrollDirection } = viewportState;
+
+  const visibleRange = getVisibleItemRange(props);
+
+  // Add overscan
+  const overscanCount = Math.ceil(overscanHeight / itemHeight);
+
+  visibleRange.startIndex = Math.max(
+    0,
+    visibleRange.startIndex -
+      (!isScrolling || scrollDirection[Axis.Y] === ScrollDirection.backward ? overscanCount : TRAILING_OVERSCAN_COUNT_WHILE_SCROLLING)
+  );
+  visibleRange.endIndex = Math.min(
+    itemCount - 1,
```

Comment:
is `endIndex` inclusive or exclusive? If you make it exclusive, maths become a little easier (end-start === count)

Reply:
Made it exclusive.